frappe/core/doctype/file/file.js

```diff
@@ -1,45 +1,72 @@
 frappe.ui.form.on("File", "refresh", function (frm) {
-	if (!frm.doc.is_folder) {
-		frm.add_custom_button(
-			__("Download"),
-			function () {
-				var file_url = frm.doc.file_url;
-				if (frm.doc.file_name) {
-					file_url = file_url.replace(/#/g, "%23");
-				}
-				window.open(file_url);
-			},
-			"fa fa-download"
-		);
-	}
+    if (!frm.doc.is_folder) {
+        frm.add_custom_button(
+            __("Download"),
+            function () {
+                var file_url = frm.doc.file_url;
+                if (frm.doc.file_name) {
+                    file_url = file_url.replace(/#/g, "%23");
+                }
+                window.open(file_url);
+            },
+            "fa fa-download"
+        );
+    }
 
-	frm.get_field("preview_html").$wrapper.html(`<div class="img_preview">
-		<img class="img-responsive" src="${frm.doc.file_url}" onerror="cur_frm.toggle_display('preview', false)" />
-	</div>`);
+    if (frappe.utils.is_image_file(frm.doc.file_url)) {
+        frm.get_field("preview_html").$wrapper.html(`<div class="img_preview">
+                                        <img class="img-responsive" src="${frm.doc.file_url}" onerror="cur_frm.toggle_display('preview', false)" />
+                                     </div>`);
+    } else if (frm.doc.file_name.split('.').pop() == 'pdf') {
+        cur_frm.toggle_display('preview', true);
```

Comment:
`cur_frm` is deprecated. `frm` is already available in the method, use it?

Should be `frm.toggle_display`